Framework/Algorithms/src/CalculateFlatBackground.cpp

```diff
@@ -38,22 +41,35 @@ void CalculateFlatBackground::init() {
   declareProperty(make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
                                                    Direction::Output),
                   "Name to use for the output workspace.");
-  auto mustHaveValue = boost::make_shared<MandatoryValidator<double>>();
-
-  declareProperty("StartX", Mantid::EMPTY_DBL(), mustHaveValue,
-                  "The X value at which to start the background fit");
-  declareProperty("EndX", Mantid::EMPTY_DBL(), mustHaveValue,
-                  "The X value at which to end the background fit");
+  std::vector<std::string> modeOptions{"Linear Fit", "Mean", "Moving Average"};
```

Comment:
We need to be careful when dealing with the property declarations as if we shift the order then it affects the Python generated `simpleapi` function.

On master this

```
python -c "import mantid.simpleapi as sapi; help(sapi.CalculateFlatBackground)"
```

gives

```
CalculateFlatBackground(InputWorkspace,StartX,EndX,WorkspaceIndexList=None,Mode=None,OutputMode=None,SkipMonitors=None,NullifyNegativeValues=None, Version=1)
```

as the function signature.

With these changes we now get

```
CalculateFlatBackground(InputWorkspace,Mode=None,StartX=None,EndX=None,AveragingWindowWidth=None,WorkspaceIndexList=None,OutputMode=None,SkipMonitors=None,NullifyNegativeValues=None, Version=1)
```

If we could guarantee that all users just use keyword arguments then this would be fine but at the moment a call such as `CalculateFlatBackground(ws, 1.0, 10.0)` in a user script would fail with these changes because the order of the arguments has changed. It's okay that `StartX` & `EndX` are now optional but we need to maintain the order of the arguments so that positional argument calls still work.